src/qgis_geonode/qgisgeonode/conf.py

```diff
@@ -0,0 +1,141 @@
+import contextlib
+import dataclasses
+import logging
+import typing
+import uuid
+
+from qgis.PyQt import QtCore
+from qgis.core import QgsSettings
+
+logger = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def qgis_settings(group_root: str):
+    """A simple context manager to help managing our own settings in QgsSettings"""
+    settings = QgsSettings()
+    settings.beginGroup(group_root)
+    try:
+        yield settings
+    finally:
+        settings.endGroup()
+
+
+@dataclasses.dataclass
+class ConnectionSettings:
+    """Helper class to manage settings for a Connection"""
+
+    id: uuid.UUID
+    name: str
+    base_url: str
+    auth_config: typing.Optional[str] = None
+
+    @classmethod
+    def from_qgs_settings(cls, connection_identifier: str, settings: QgsSettings):
+        return cls(
+            id=uuid.UUID(connection_identifier),
```

Comment:
we have to check for `ValueError('badly formed hexadecimal UUID string')` here

Reply:
I don't seem to have that error.

can you please try to empty your relevant `QgsSettings` and try again? Try this in the QGIS Python console:

```python
from qgis_geonode.qgisgeonode.conf import connections_manager
connections_manager.delete_all_connections()
connections_manager.clear_current_connection()
```

then try again and see if you get that error. If you do I will fix it, but I suspect it is not going to happen anymore

Reply:
ok, let me try it.

Reply:
yes, the error disappeared!